examples/chip-tool-darwin/commands/common/CHIPCommandStorageDelegate.mm

```diff
@@ -27,21 +27,25 @@ @implementation CHIPToolPersistentStorageDelegate
 
 // MARK: CHIPPersistentStorageDelegate
 
-- (NSString *)CHIPGetKeyValue:(NSString *)key
+- (nullable NSData *)valueForKey:(NSString *)key
 {
-    NSString * value = CHIPGetDomainValueForKey(kCHIPToolDefaultsDomain, key);
+    NSData * value = CHIPGetDomainValueForKey(kCHIPToolDefaultsDomain, key);
     NSLog(@"CHIPPersistentStorageDelegate Get Value for Key: %@, value %@", key, value);
     return value;
 }
 
-- (void)CHIPSetKeyValue:(NSString *)key value:(NSString *)value
+- (BOOL)setValue:(NSData *)value forKey:(NSString *)key
 {
-    CHIPSetDomainValueForKey(kCHIPToolDefaultsDomain, key, value);
+    return CHIPSetDomainValueForKey(kCHIPToolDefaultsDomain, key, value);
 }
 
-- (void)CHIPDeleteKeyValue:(NSString *)key
+- (BOOL)removeValueForKey:(NSString *)key
 {
+    if (CHIPGetDomainValueForKey(kCHIPToolDefaultsDomain, key) == nil) {
```

Comment:
Maybe we should just call this once at the top and then `return key != nil`

Reply:
So like this:
```
    NSData * oldValue = CHIPGetDomainValueForKey(kCHIPToolDefaultsDomain, key);
    CHIPRemoveDomainValueForKey(kCHIPToolDefaultsDomain, key);
    return oldValue != nil;
```
?